actionmailer/test/log_subscriber_test.rb

```diff
@@ -35,6 +41,29 @@ def test_deliver_is_notified
     BaseMailer.deliveries.clear
   end
 
+  def test_deliver_is_notified_when_exception_is_raised
+    @original_delivery_method = ActionMailer::Base.delivery_method
+    ActionMailer::Base.delivery_method = ExceptionDelivery
+
+    mail = BaseMailer.welcome
+    assert_raise Net::SMTPAuthenticationError do
+      mail.deliver_now
+    end
+    wait
+
+    assert_equal(0, @logger.logged(:info).size)
+
+    assert_equal(1, @logger.logged(:error).size)
+    assert_match(/Net::SMTPAuthenticationError was raised when sending mail to system@test.lindsaar.net/, @logger.logged(:error).first)
```

Comment:
Hey @cirojr it's really awesome that you want to contribute. Regarding using this code; before it's merged into Rails, you could fork and cherrypick this commit, or you could link right to the branch like:

`gem 'rails', github: 'gnatiusreza/rails', branch: 'action_mailer_log_subscriber_on_exception'

Since everyone who is on this PR gets an email every time you comment this isn't a good place to discuss contributing. You can DM me on twitter (same handle here @eileencodes) if you want to chat more about contributing. I'd be happy to point you in the right direction. 😄